lib/ansible/plugins/callback/json.py

```diff
@@ -69,6 +69,7 @@ def _new_task(self, task):
             'task': {
                 'name': task.get_name(),
                 'id': str(task._uuid),
+                'action': task.action,
```

Comment:
The problem with this, is that the action could be templated and be different based on a host.

```
- action: "helloworld_{{ ansible_system|lower }}"
```
This is why currently the action is only inserted into results and not in the task definition itself.

Reply:
Thanks for the hint, I didn't know about that 😃  So, this is just missing in Zuul's json callback plugin and I will add it there.